internal/reconciler/shoot/reconciler.go

```diff
@@ -120,29 +128,49 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{}, nil
 }
 
-func (r *Reconciler) deleteOIDC(ctx context.Context, log logr.Logger, shoot *gardencorev1beta1.Shoot) (ctrl.Result, error) {
+// handleDeletion handles the deletion of a shoot and its associated OIDC resource
+func (r *Reconciler) handleDeletion(ctx context.Context, log logr.Logger, shoot *gardencorev1beta1.Shoot) (ctrl.Result, error) {
+	if !controllerutil.ContainsFinalizer(shoot, FinalizerName) {
+		log.Info("Finalizer not present, nothing to clean up")
+		return reconcile.Result{}, nil
+	}
```

Comment:
Even if finalizer is not present we should still try to cleanup the oidc resource. I think we can completely skip this check 🤔